runtime/criusupport/criusupport.c

```diff
@@ -298,48 +279,48 @@ Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
 		vmFuncs->acquireExclusiveVMAccess(currentThread);
 
 		if (FALSE == vmFuncs->jvmCheckpointHooks(currentThread)) {
-			resultType = vm->criuSupportJVMCheckpointFailure;
+			currentExceptionClass = vm->criuJVMCheckpointExceptionClass;
+			// TODO: error message
```

Comment:
@tajila not sure about these error messages. Also line 299 and 343

Reply:
For these, we can assume that the call will set an exception. You can then wrap that exception with a checkpoint exception. `setCurrentExceptionWithUtfCause`

Reply:
Line 343 is for the UnsupportedOperationException. Should I remove that and the `if (vmFuncs->isCheckpointAllowed(currentThread)) {` on line 195 (since the java code already tests `isCRIUSupportEnabled()`)?